### DISCUSSION TITLE
Tap Hold による Layer 切り替え直後のクリックが機能しないことがある

### DISCUSSION THREAD

**Discussion by mtsmfm** [Yowkees/keyball#54] [Q&A] [answered]
デフォルトだと、 左手親指の左から2番目のキーとJキーの組み合わせでクリックになるかと思うのですが、左手親指の左から2番目のキーを押した直後にJキーを押すとクリックもスペースも両方とも反応しないことがあります。
Tap Hold ではなく普通の Layer 切り替えキー設定にすると反応するので、半田付けの問題ではないと考えています。
keyball のファームではなく QMK 自体の問題という可能性もありますが、まずは他にも同様の方がいるかを知りたいです。

**Comment by Yowkees** [comment]
QMKファームウェアにて、何mS長押しされた場合にHoldと判定するかのパラメータがあります。
そのため、レイヤーキーを押した直後にクリックキーを押すと認識されないようです。
気になる場合は、この数値を短めに調整する必要があります。

**Reply by mtsmfm** [reply]
@Yowkees 回答ありがとうございます。
Yowkees さんも意図的に素早い操作をすると同様の症状が再現できるということでよろしいでしょうか。
Tap Hold による Layer 切り替えでも、クリック以外だと比較的スムーズに反応する気がしており、QMK の問題なのか自分だけの問題なのかどうかが気になっております。

**Reply by koron** [reply, marked as answer]
このあたり (`TAPPING_TERM`) ですかね。
デフォルトでは200msに設定されているそうです。

https://beta.docs.qmk.fm/using-qmk/software-features/tap_hold#tapping-term
